docs/azure/integrations-overview.md

```diff
@@ -40,6 +64,119 @@ The preceding code:
 
+## Mark Azure resources as existing
+
+In addition to the pre-existing <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.AddConnectionString*> API, .NET Aspire provides expanded support for referencing existing Azure resources. This is achieved through the `PublishAsExisting`, `RunAsExisting`, and `AsExisting` APIs. These APIs allow developers to reference already-deployed Azure resources, configure them, and generate appropriate deployment manifests using Bicep templates.
+
+### Configure existing Azure resources for run mode
+
+The `RunAsExisting` method is used when a distributed application is executing in "run" mode. In this mode, it assumes that the referenced Azure resource already exists and integrates with it during execution without provisioning the resource. To mark an Azure resource as existing, call the `RunAsExisting` method on the resource builder. Consider the following example:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder();
+
+var existingResourceName = builder.AddParameter("existingResourceName");
+
+var serviceBus = builder.AddAzureServiceBus("messaging")
+    .RunAsExisting(existingResourceName)
+    .WithQueue("queue");
+```
+
+The preceding code:
+
+- Creates a new `builder` instance.
+- Adds a parameter named `existingResourceName` to the builder.
+- Adds an Azure Service Bus resource named `messaging` to the builder.
+- Calls the `RunAsExisting` method on the `serviceBus` resource builder, passing the `existingResourceName` parameter—alternatively, you can use the `string` parameter overload.
+- Adds a queue named `queue` to the `serviceBus` resource.
+
+### Configure existing Azure resources for publish mode
+
+The `PublishAsExisting` method is used in "publish" mode when the intent is to declare and reference an already-existing Azure resource during publish mode. This API facilitates the creation of manifests and templates that include resource definitions that map to existing resources in Bicep.
+
+To mark an Azure resource as existing in for the "publish" mode, call the `PublishAsExisting` method on the resource builder. Consider the following example:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder();
+
+var existingResourceName = builder.AddParameter("existingResourceName");
+
+var serviceBus = builder.AddAzureServiceBus("messaging")
+    .PublishAsExisting(existingResourceName)
+    .WithQueue("queue");
```

Comment:
```suggestion
var serviceBus = builder.AddAzureServiceBus("messaging")
    .PublishAsExisting(existingResourceName);
```

WithQueue is obsolete in 9.1

Reply:
If you want to convert it, use:

```
var serviceBus = builder.AddAzureServiceBus("messaging")
    .PublishAsExisting(existingResourceName);

serviceBus.AddServiceBusQueue("queue");
```

AddServiceBusQueue returns the queue resource. (Just like AddDatabase returns the child database resource)

Reply:
See https://github.com/dotnet/docs-aspire/issues/2327#issuecomment-2673088907

Reply:
Good catch, thanks for helping out with this.